packages/bpk-component-banner-alert/src/withNormalBehaviour.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const withNormalBehaviour = (WrappedComponent) => {
+  class component extends Component {
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        expanded: false,
+        dismissed: false,
+      };
+
+      this.onExpandToggle = this.onExpandToggle.bind(this);
+      this.onDismiss = this.onDismiss.bind(this);
+    }
+
+    componentWillMount() {
+      const { hideAfter } = this.props;
+
+      if (hideAfter && hideAfter > 0) {
+        this.hideIntervalId = setTimeout(() => {
+          this.onDismiss();
+        }, hideAfter * 1000);
+      }
+    }
+
+    componentWillUnmount() {
+      if (this.hideIntervalId) {
+        clearTimeout(this.hideIntervalId);
+      }
+    }
+
+    onExpandToggle() {
+      this.setState(state => ({
+        expanded: !state.expanded,
+      }));
+    }
+
+    onDismiss() {
+      this.setState({ dismissed: true });
+    }
+
+    render() {
+      const {
+        onDismiss, expanded, show, hideAfter, ...rest
+      } = this.props;
+
+      return this.state.dismissed ? null : (
+        <WrappedComponent
+          expanded={this.state.expanded}
+          onExpandToggle={this.onExpandToggle}
+          onDismiss={this.onDismiss}
+          show
+          {...rest}
+        />
+      );
+    }
+  }
```

Comment:
We should set the componentName for this. See https://github.com/Skyscanner/backpack/blob/73b56b9840139fccad7b6f65dd66f0c964c189de/packages/bpk-scrim-utils/src/withScrim.js#L158

Reply:
I didn't knew about this, nice! I'll add it to our squad apps :-P

Added here